algoliasearch/client_test.go

```diff
@@ -430,3 +432,75 @@ func TestDnsTimeout(t *testing.T) {
 		t.Fatalf("TestDnsTimeout: Spent %d seconds instead of <5s to perform the 10 retries", int(delta.Seconds()))
 	}
 }
+
+func TestMultiClusterManagement(t *testing.T) {
+	client := initMCMClient(t)
+	var uniqueUserID = fmt.Sprintf("go-MCM-tests-%d", rand.Int())
```

Comment:
Why do we need to introduce randomness here?

Reply:
Note that I'm not super familiar with MCM and its userIDs.

Reply:
While developing, tests would often fail and relaunching would also fail because of passed userID already existing ^^
It's not so important once everything is stable but it really helps you have "operation pending on userId XX errors".

Reply:
I don't know if it's possible with MCM but what I'm doing for all the other tests is that I force clean/remove all the possible resources I'm gonna use (usually I delete the index and recreate it right after for instance) so that I'm 100% sure my test is always running the same, whether or not there were leftovers from previous tests. WDYT?